src/settings/global_user.rs

```diff
@@ -17,7 +26,90 @@ pub enum GlobalUser {
 
 impl GlobalUser {
     pub fn new() -> Result<Self, failure::Error> {
-        get_global_config()
+        let environment = Environment::with_whitelist(ENV_VAR_WHITELIST.to_vec());
+
+        let config_path = default_config_file().expect("could not find global config directory");
+
+        GlobalUser::build(environment, config_path)
+    }
+
+    fn build<T: 'static + QueryEnvironment>(
+        environment: T,
+        config_path: PathBuf,
+    ) -> Result<Self, failure::Error>
+    where
+        T: config::Source + Send + Sync,
+    {
+        if let Some(user) = Self::from_env(environment) {
+            user
+        } else {
+            Self::from_file(config_path)
+        }
```

Comment:
i want to make sure that this does not change previous behavior - did we ever allow reading in an email from an environment variable and an api key from the config file? perhaps someone just wants to override one or the other. not sure if we previously supported that or not (i think we probably did?)

Reply:
i'm not saying we _should_ but we might have, and this would be a breaking change if that were the case

Reply:
we did; however this is the behavior we landed on in the discussion in #834